code/go/0chain.net/smartcontract/dbs/event/miner.go

```diff
@@ -72,11 +72,16 @@ func (edb *EventDb) GetMinerWithDelegatePools(id string) (Miner, []DelegatePool,
 		return m, nil, fmt.Errorf("mismatched miner; want id %s but have id %s", id, minerDps[0].Miner.ID)
 	}
 	m = minerDps[0].Miner
-	if id != minerDps[0].ProviderRewards.ProviderID {
-		return m, nil, fmt.Errorf("mismatched miner; want id %s but have id%s in provider rewrards",
-			id, minerDps[0].Miner.ID)
-	}
+
+	//if "" == minerDps[0].ProviderRewards.ProviderID {
+	//	return m, nil, fmt.Errorf("cannot find provider rewards table for miner %v. rewards %v",
+	//		id, minerDps[0].ProviderRewards)
+	//} todo remove
 	m.Rewards = minerDps[0].ProviderRewards
+	m.Rewards.ProviderID = id
+	if len(minerDps) == 1 && minerDps[0].DelegatePool.ProviderID == "" {
```

Comment:
why is it possible to have empty provider id?

Reply:
I don't know. I am just fixing symptoms at this point.

Reply:
I would spent some time to investigate how could the ProviderReward to be empty if it should not be, otherwise we would just hide the issue and if other APIs or functions need to touch the data, they will get invalid data too. Anyway, I checked how the data went to ProviderRewards table, and then found this https://github.com/0chain/0chain/issues/2055. Please have a check and fix.

Reply:
@peterlimg  https://github.com/0chain/0chain/issues/2055 I don't think this is a problem. The Global Node values are read from [readConfig](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/minersc/models.go#L261) In particular, the GlobalNode.BlockReward is multiplied by 1e10 using ParseZCN, so as far as I can tell the correct value is calculated in `DistributeRewardRandN`.

Reply:
@Sriep I left a comment in issue #2055, it might be able to fix the issue.